src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/SetSettlementAgreementDeadlineCallbackHandler.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.hmcts.reform.civil.handler.callback.user;
+
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.SET_SETTLEMENT_AGREEMENT_DEADLINE;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+import uk.gov.hmcts.reform.civil.callback.Callback;
+import uk.gov.hmcts.reform.civil.callback.CallbackHandler;
+import uk.gov.hmcts.reform.civil.callback.CallbackParams;
+import uk.gov.hmcts.reform.civil.callback.CaseEvent;
+
+import java.util.Collections;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.service.DeadlinesCalculator;
+
+@Service
+@RequiredArgsConstructor
+public class SetSettlementAgreementDeadlineCallbackHandler extends CallbackHandler {
+
+    private static final List<CaseEvent> EVENTS = Collections.singletonList(SET_SETTLEMENT_AGREEMENT_DEADLINE);
+    private final DeadlinesCalculator deadlinesCalculator;
+    private final ObjectMapper objectMapper;
+
+    @Override
+    protected Map<String, Callback> callbacks() {
+        return new ImmutableMap.Builder<String, Callback>()
+            .put(callbackKey(ABOUT_TO_SUBMIT), this::setSettlementAgreementDeadline)
```

Comment:
this can be moved outside of function and return instance of Map in callback function

Reply:
Thanks, will leave because this is part of changes for CIV-11543 which was already reviewed and now in Deploy Awaits.